src/transaction/log_manager.c

```diff
@@ -14684,6 +14733,441 @@ cdc_is_filtered_user (char *user)
+
+      if (trid == context->trid && log_type == LOG_SUPPLEMENTAL_INFO)
+	{
+	  LOG_REC_SUPPLEMENT *supplement;
+	  int supplement_length;
+	  SUPPLEMENT_REC_TYPE rec_type;
+
+	  OID classoid;
+
+	  LOG_LSA undo_lsa, redo_lsa;
+
+	  /* TODO: 
+	   * modulize the process of getting supplement data 
+	   * */
+
+	  LOG_READ_ADVANCE_WHEN_DOESNT_FIT (thread_p, sizeof (*supplement), &process_lsa, log_page_p);
+
+	  supplement = (LOG_REC_SUPPLEMENT *) (log_page_p->area + process_lsa.offset);
+	  supplement_length = supplement->length;
+	  rec_type = supplement->rec_type;
+
+	  LOG_READ_ADD_ALIGN (thread_p, sizeof (*supplement), &process_lsa, log_page_p);
+
+	  if (cdc_get_undo_record (thread_p, log_page_p, cur_log_rec_lsa, &supp_recdes) != S_SUCCESS)
+	    {
+	      /* TODO:
+	       * after refactor cdc_get_undo_record, then remove error setting
+	       * modify cdc_get_undo_record() to return error, not scan code 
+	       * */
+	      error = ER_FAILED;
+	      goto error;
+	    }
+
+	  supplement_length = sizeof (supp_recdes.type) + supp_recdes.length;
+
+	  if (supplement_length > supplement_alloc_length)
+	    {
+	      char *tmp = (char *) db_private_realloc (thread_p, supplement_data, supplement_length);
+	      if (tmp == NULL)
+		{
+		  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, supplement_length);
+		  error = ER_OUT_OF_VIRTUAL_MEMORY;
+		  goto error;
+		}
+	      else
+		{
+		  supplement_data = tmp;
+		  supplement_alloc_length = supplement_length;
+		}
+	    }
+
+	  /* will be replaced with CDC_MAKE_SUPPLEMENT_DATA */
+	  memcpy (supplement_data, &supp_recdes.type, sizeof (supp_recdes.type));
+	  memcpy (supplement_data + sizeof (supp_recdes.type), supp_recdes.data, supp_recdes.length);
+
+	  free_and_init (supp_recdes.data);
+
+	  switch (rec_type)
+	    {
+	    case LOG_SUPPLEMENT_INSERT:
+	    case LOG_SUPPLEMENT_TRIGGER_INSERT:
+	      memcpy (&classoid, supplement_data, sizeof (OID));
+
+	      /* classoid filter */
+	      if (context->classoid_set.find (classoid) == context->classoid_set.end ())
+		{
+		  break;
+		}
+
+	      memcpy (&redo_lsa, supplement_data + sizeof (OID), sizeof (LOG_LSA));
+
+	      error = cdc_get_recdes (thread_p, NULL, NULL, &redo_lsa, &redo_recdes, true);
+	      if (error != NO_ERROR)
+		{
+		  goto error;
+		}
+
+	      log_info_entry = (CDC_LOGINFO_ENTRY *) db_private_alloc (thread_p, sizeof (CDC_LOGINFO_ENTRY));
+	      if (log_info_entry == NULL)
+		{
+		  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, sizeof (CDC_LOGINFO_ENTRY));
+		  error = ER_OUT_OF_VIRTUAL_MEMORY;
+		  goto error;
+		}
+
+	      error =
+		cdc_make_dml_loginfo (thread_p, trid, context->user,
+				      rec_type == LOG_SUPPLEMENT_INSERT ? CDC_INSERT : CDC_TRIGGER_INSERT, classoid,
+				      NULL, &redo_recdes, log_info_entry, true);
+
+	      if (error != NO_ERROR)
+		{
+		  goto error;
+		}
+
+              // *INDENT-OFF*
```

Comment:
It needs `// *INDENT-OFF*`?

Reply:
Oh.. no, I will remove it. At first, I thought I need to `INDENT-OFF` to c++ functions.